### PR TITLE
[DON-2384] Add Unicode icon detector, documentation, and demo fixes

### PR DESCRIPTION
**Pull request by LokmaneKrizou**
## Summary

**Unicode Icon Detection:**
- Add `UnicodeIconUsageDetector`: Detects direct Unicode icon usage
- Add `UnicodeIconTokenMap`: Maps Unicode icons to Backpack icons
- Add unit test for Unicode icon detection

**Documentation:**
- Add `LINT_GUIDE.md`: Comprehensive guide for lint rules
- Add `README.md`: Overview of backpack-lint module
- Add `docs/LINT_FIXES.md`: How to fix lint issues
- Add `docs/LINT_PATTERNS.md`: Common patterns and anti-patterns
- Add `docs/LINT_TESTING.md`: How to test lint rules

**Improvements:**
- Update `BpkComposeComponentUsageDetector` with TEXT and LINK components
- Fix formatting in `BpkComponentUsageDetector`
- Fix hardcoded colors in `ColorsStory` demo
- Update `.gitignore`

**This is PR 5 of 5 - depends on PR 1 & 2. Can be merged in parallel with PR 3 & 4.**

## Dependency Graph

```
PR 1 (Token Gen)
    ↓
PR 2 (Utilities)
    ↓
    ├── PR 3 (Color/Typography)
    ├── PR 4 (Dimensions)
    └── PR 5 (Unicode/Docs) ← YOU ARE HERE
```

## Depends On

- #2578
- #2579

## Files Changed

- `backpack-lint/src/main/java/net/skyscanner/backpack/lint/check/UnicodeIconUsageDetector.kt`
- `backpack-lint/src/main/java/net/skyscanner/backpack/lint/check/UnicodeIconTokenMap.kt`
- `backpack-lint/LINT_GUIDE.md`, `README.md`, `docs/*.md`
- `app/src/main/java/net/skyscanner/backpack/demo/compose/ColorsStory.kt`
- `.gitignore`

## Test plan

- [ ] Run `./gradlew :backpack-lint:test` to verify all unit tests pass
- [ ] Test Unicode icon detector on sample code
- [ ] Review documentation for accuracy and completeness

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: backpack-lint/src/main/java/net/skyscanner/backpack/lint/check/UnicodeIconUsageDetector.kt [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import com.intellij.psi.PsiMethod
+import org.jetbrains.uast.UCallExpression
+import org.jetbrains.uast.ULiteralExpression
+import org.jetbrains.uast.UPolyadicExpression
+
+@Suppress("UnstableApiUsage")
+class UnicodeIconUsageDetector : Detector(), SourceCodeScanner {
+
+    companion object {
+        private val EXPLANATION =
```
Comment (RichSutherSky) [inline backpack-lint/src/main/java/net/skyscanner/backpack/lint/check/UnicodeIconUsageDetector.kt]: NP const
